docs/proposals/job.md

```diff
@@ -0,0 +1,133 @@
+
+
+## Motivation
+Jobs are needed for executing multi-pod computation to completion; a good example
+here would be the ability to implement any type of batch oriented tasks.
+
+
+## Implementation
+Job controller is similar to replication controller in that they manage pods.
+This implies they will follow the same controller framework that replication
+controllers already defined.  The biggest difference between a `Job` and a
+`ReplicationController` object is the purpose; `ReplicationController`
+ensures that a specified number of Pods are running at any one time, whereas
+`Job` is responsible for keeping the desired number of Pods to a completion of
+a task.  This difference will be represented by the `RestartPolicy` which is
+required to always take value of `RestartPolicyNever` or `RestartOnFailure`.
+
+
+The new `Job` object will have the following content:
+
+```go
+// Job represents the configuration of a single job.
+type Job struct {
+    TypeMeta
+    ObjectMeta
+
+    // Spec is a structure defining the expected behavior of a job.
+    Spec JobSpec
+
+    // Status is a structure describing current status of a job.
+    Status JobStatus
+}
+
+// JobList is a collection of jobs.
+type JobList struct {
+    TypeMeta
+    ListMeta
+
+    Items []Job
+}
+```
+
+`JobSpec` structure is defined to contain all the information how the actual job execution
+will look like.
+
+```go
+// JobSpec describes how the job execution will look like.
+type JobSpec struct {
+
+    // MaxParallelism specifies the desired number of pods the job is run at
+    // any one time.
+    MaxParallelism int
+
+    // Completions specifies the desired number of successfully finished pods the
+    // job should be run with.
+    Completions int
+
+    // Optional duration in seconds relative to the StartTime that the job may be active
+    // before the system actively tries to terminate it; value must be positive integer
+    ActiveDeadlineSeconds *int64
+
+    // Selector is a label query over pods that should match the pod count.
+    Selector map[string]string
+
+    // Template is the object that describes the pod that will be created when
+    // executing a job.
+    Template *PodTemplateSpec `json:"template,omitempty"`
+}
+```
+
+`JobStatus` structure is defined to contain informations about pods executing
+specified job.  The structure holds information about pods currently executing
+the job.
+
+```go
+// JobStatus represents the current state of a Job.
+type JobStatus struct {
+
+    // Active is the number of actively running pods.
+    Active int
+
+    // Successful is the number of pods successfully completed their job.
+    Successful int
+
+    // Unsuccessful is the number of pods failures, this applies only to jobs
+    // created with RestartPolicyNever, otherwise this value will always be 0.
+    Unsuccessful int
+}
+
+// JobExecution represents the current state of a single execution of a Job.
+type JobExecution struct {
```

Comment:
Agree with @derekwaynecarr that inert object can make `Job.ObjectMeta.CreationTimestamp` not equal to the "start of execution of the job".  

Also, people might intentionally write a batch queueing system that scales Jobs to 0 as a way of having them "queue", or as a way to pause a "low priority job" while a "high priority job" runs and consumes resources.

Even without an explicit batch queuing system, the cluster might be short on resources, and so some or all of the job's pods might remain unscheduled until there is space.

  And once we add QoS, if jobs have a request << limit, then they make take a variable amount of wall time to complete.  

I suggest dropping `JobExecution` for now, or else giving more detail on the use cases so that we can make sure we have the right definition of a deadline.